### PR TITLE
updating original object uids in register_all

### PR DESCRIPTION
**Pull request by ventura-rivera**
# Citrine Python PR

## Description 
Please briefly explain the goal of the changes/this PR.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

**Comment by ventura-rivera** [comment]
> Is there a way to add even a basic citrine-python test?

I'm looking through the factory objects to see how I can do this

File: src/citrine/resources/dataset.py [hunk 1]
```diff
@@ -231,9 +231,14 @@ def register_all(self, data_concepts_resources: List[ResourceType]) -> List[Reso
 
         :return the registered versions
         """
-        return [self.register(resource) for resource
```
Comment (maxhutch) [inline src/citrine/resources/dataset.py:234]: Is this change necessary?  I would have thought that `self.register(resource)` would return the resource with the uid set.

File: tests/utils/session.py [hunk 2]
```diff
@@ -86,12 +86,12 @@ def checked_delete(self, path: str, **kwargs) -> dict:
         self.calls.append(FakeCall('DELETE', path, params=kwargs.get('params')))
         return self._get_response()
 
-    def _get_response(self):
+    def _get_response(self, default_response: dict = {}):
```
Comment (dekatzenel) [inline tests/utils/session.py:89]: python has a thing about default mutable arguments: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

File: src/citrine/resources/dataset.py (hunk 1 above)
Comment (dekatzenel) [reply src/citrine/resources/dataset.py:234]: More specifically, the issue occurs when trying to store an object that has a reference to an object that doesn't have a uid

File: src/citrine/resources/dataset.py [hunk 3]
```diff
@@ -231,9 +232,14 @@ def register_all(self, data_concepts_resources: List[ResourceType]) -> List[Reso
...
-                in (sorted(data_concepts_resources,
-                           key=lambda resource: writable_sort_order(resource.typ)))]
+        resources = list()
+        for resource in (sorted(data_concepts_resources,
+                         key=lambda resource: writable_sort_order(resource.typ))):
+            registered_resource = self.register(resource)
+            if isinstance(registered_resource, BaseEntity):
+                resource.uids = registered_resource.uids
```
Comment (dekatzenel) [inline src/citrine/resources/dataset.py:240]: Updating uids in place is unexpected behavior that is worth documenting on the method (including a quick explanation of what this supports)

File: tests/utils/session.py (hunk 2 above)
Comment (ventura-rivera) [reply tests/utils/session.py]: Fixed this (used the pattern in the link)

File: src/citrine/resources/dataset.py (hunk 3 above)
Comment (ventura-rivera) [reply src/citrine/resources/dataset.py:243]: Added explanation on behavior to the docstrings

**Comment by ventura-rivera** [comment]
@dekatzenel Yes, I can do that
